src/EventStore.LogV3/Raw.cs

```diff
@@ -72,8 +80,21 @@ public struct StreamWriteId {
 			}
 
 			public long StartingEventNumber {
-				get => _startingEventNumber;
-				set => _startingEventNumber = value;
+				// little-endian. must not write the 2 MOST significant bytes
+				get {
+					var x = _startingEventNumber & SixLeastSignificantBytes;
+					return x == SixLeastSignificantBytes ? EventNumberDeletedStream : x;
+				}
+				set {
+					if (value == EventNumberDeletedStream) {
+						_startingEventNumber |= SixLeastSignificantBytes;
```

Comment:
This one's ok since they are all ones being ORed anyway but for clarity it could be changed to:
_startingEventNumber = (_startingEventNumber & ~SixLeastSignificantBytes) | SixLeastSignificantBytes;